### PR TITLE
test(sidebar): add cypress tests

### PR DESCRIPTION
**Pull request by stephenogorman**
### Proposed behaviour

Add Cypress tests for the `Sidebar` component to use new cypress-component-react framework for testing.
Add `sidebar.test.js`. 

### Current behaviour

Current process is the regular Cypress approach with BDD

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- [ ] Run `npx cypress open --component` to check if there is newly added test.file
- [ ] Check if the `sidebar.test.js` file passed
- [ ] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [ ] Run `npx cypress run --e2e` to check none of the feature files have regressed

### CODE REVIEW

File: src/components/sidebar/sidebar.test.js [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+import Toast from "../toast";
+import Typography from "../typography";
+import { keyCode } from "../../../cypress/support/helper";
+import { CHARACTERS } from "../../../cypress/support/component-helper/constants";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+
+const SidebarComponent = ({ ...props }) => {
+  return (
+    <>
+      <Button onClick={function noRefCheck() {}}>Open sidebar</Button>
```
Comment (DlgSHi) [inline src/components/sidebar/sidebar.test.js:22]: do we need this `<Button>` here?

File: src/components/sidebar/sidebar.test.js [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+    );
+
+    it.each([
+      ["left", 0, 852],
+      ["right", 852, 0],
+    ])("verify Sidebar position is %s", (boolVal, left, right) => {
+      CypressMountWithProviders(<SidebarComponent position={boolVal} />);
+
+      sidebarPreview().should("have.css", "left", `${left}px`);
+      sidebarPreview().should("have.css", "right", `${right}px`);
```
Comment (DlgSHi) [inline src/components/sidebar/sidebar.test.js:121]:
```suggestion
      .and("have.css", "right", `${right}px`);
```

File: src/components/sidebar/sidebar.test.js [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+      );
+
+      sidebarPreview().should(
+        "have.attr",
+        "aria-describedby",
+        CHARACTERS.STANDARD
+      );
+    });
+
+    it("verify Sidebar has aria-label 'cypress-data'", () => {
```
Comment (DlgSHi) [inline src/components/sidebar/sidebar.test.js:136]:
```suggestion
    it("verify Sidebar has aria-label cypress-data", () => {
```

File: src/components/sidebar/sidebar.test.js [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+      sidebarPreview().should(
+        "have.attr",
+        "aria-labelledby",
+        CHARACTERS.STANDARD
+      );
+    });
+
+    it.each([
+      ["extra-small", 214],
```
Comment (DlgSHi) [inline src/components/sidebar/sidebar.test.js:157]: can we use here `src/components/sidebar/sidebar.config.ts` ?

File: src/components/sidebar/sidebar.test.js [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+        .children()
+        .should("have.attr", "data-component", "sidebar-header");
+      sidebarPreview()
+        .children()
+        .children()
+        .eq(0)
+        .and("have.text", "Sidebar Header");
+    });
+
+    it("verify Sidebar has role 'cypress-data'", () => {
```
Comment (DlgSHi) [inline src/components/sidebar/sidebar.test.js:187]:
```suggestion
    it("verify Sidebar has role cypress-data", () => {
```

File: src/components/sidebar/sidebar.test.js [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+      CypressMountWithProviders(
+        <SidebarComponent role={CHARACTERS.STANDARD} />
+      );
+
+      sidebarPreview().should("have.attr", "role", CHARACTERS.STANDARD);
+    });
+
+    it("verify Sidebar has data-element 'cypress-data'", () => {
```
Comment (DlgSHi) [inline src/components/sidebar/sidebar.test.js:195]:
```suggestion
    it("verify Sidebar has data-element cypress-data", () => {
```

File: src/components/sidebar/sidebar.test.js [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+
+      sidebarPreview()
+        .parent()
+        .parent()
+        .parent()
+        .parent()
+        .should("have.attr", "data-element", CHARACTERS.STANDARD);
+    });
+
+    it("verify Sidebar has data-role 'cypress-data'", () => {
```
Comment (DlgSHi) [inline src/components/sidebar/sidebar.test.js:208]:
```suggestion
    it("verify Sidebar has data-role cypress-data", () => {
```

File: src/components/sidebar/sidebar.test.js [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+        .parent()
+        .parent()
+        .parent()
+        .should("have.attr", "data-role", CHARACTERS.STANDARD);
+    });
+
+    it.each([
+      [true, "not.be.visible"],
+      [false, "be.visible"],
+    ])("verify Sidebar with disableEscKey prop %s", (boolVal, state) => {
```
Comment (DlgSHi) [inline src/components/sidebar/sidebar.test.js:224]:
```suggestion
    ])("verify Sidebar with disableEscKey prop set to %s", (boolVal, state) => {
```

File: src/components/sidebar/sidebar.test.js [hunk 9]
```diff
@@ -0,0 +1,269 @@
...
+      CypressMountWithProviders(<SidebarComponent disableEscKey={boolVal} />);
+
+      sidebarPreview().trigger("keydown", keyCode("Esc"));
+      sidebarPreview().should(state);
+    });
+
+    it.each([
+      [true, "exist"],
+      [false, "not.exist"],
+    ])("verify Sidebar with open prop %s", (boolVal, state) => {
```
Comment (DlgSHi) [inline src/components/sidebar/sidebar.test.js:234]:
```suggestion
    ])("verify Sidebar with open prop set to %s", (boolVal, state) => {
```

File: src/components/sidebar/sidebar.test.js [hunk 10]
```diff
@@ -0,0 +1,269 @@
...
+      getComponent("toast").children().eq(2).children().click();
+      getComponent("toast").should("not.exist");
+    });
+  });
+
+  describe("check events for Sidebar component", () => {
+    let callback;
+
+    beforeEach(() => {
+      callback = cy.stub();
```
Comment (DlgSHi) [inline src/components/sidebar/sidebar.test.js:255]: do we need this `describe` and `before` for only 1 test?

File: src/components/sidebar/sidebar.test.js [hunk 11]
```diff
@@ -0,0 +1,269 @@
...
+    });
+
+    it("should call onClick callback when a click event is triggered", () => {
```
Comment (DlgSHi) [inline src/components/sidebar/sidebar.test.js:258]:
```suggestion
    it("should call onCancel callback when a click event is triggered", () => {
```

**Review by DlgSHi — CHANGES_REQUESTED** [review]
focusableSelectors prop is not covered

**Review by DlgSHi — CHANGES_REQUESTED** [review]
Still no tests for `focusableSelectors` prop

**Review by DlgSHi — APPROVED** [review]
LGTM, only squash the commits

**Review by DianeGit22 — APPROVED** [review]

**Pull request merged**